CMake/FindMKL.cmake

```diff
@@ -2,42 +2,105 @@
 # This needs a lot of work to make it robust
 INCLUDE( CheckCXXSourceCompiles )
 
+if ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
+  # if not intel and MKL_ROOT not set
+  if (NOT MKL_ROOT)
+    find_path(MKL_ROOT "mkl.h"
+      HINTS "${MKLROOT} ${MKL_HOME} $ENV{MKLROOT} $ENV{MKL_ROOT} $ENV{MKL_HOME}")
```

Comment:
I think HINTS expects multiple separated arguments following it, not a single argument (no quotes around all of the variables).
Also, it only seemed to work when the /include was explicitly added (${MKLROOT}/include)

Reply:
Right on the expansion. And the PATH_SUFFIXES dropped out.  Which is why that replace is there.
I don't need this but this should cover some of the mkl modules I've seen, thought it would be nice to start moving in that direction